crates/benchmark-runner/src/guest_programs.rs

```diff
@@ -0,0 +1,105 @@
+//! Guest program input generation and metadata types
+
+use guest_libs::BincodeBlock;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::path::Path;
+use walkdir::WalkDir;
+use witness_generator::BlockAndWitness;
+use zkvm_interface::Input;
+
+/// Metadata trait for guest program inputs
+pub trait GuestInputMetadata: Serialize + DeserializeOwned + Clone + Send + Sync {}
+impl GuestInputMetadata for () {}
+
+/// Represents a guest program input with associated metadata
+#[derive(Debug, Clone)]
+pub struct GuestInput<M: GuestInputMetadata> {
+    /// The name of the guest program input.
+    pub name: String,
+    /// The standard input to be provided to the guest program.
+    pub stdin: Input,
+    /// Associated metadata for the guest program input.
+    pub metadata: M,
+}
```

Comment:
This is the main struct that each guest program type will generate for the main benchmark-runner.

As in, the benchmark runner receives a list of this type and runs the benchmark:
- The `name` which will be used for saving the report.
- The `Input` which is passed as stdin to the underlying zkVM.
- A `metadata` generic struct that each guest program can attach as extra info. For example, `stateless-validator` has the known `block_gas_used : u64`; the `empty-program` doesn't add anything, and `rlp-encoding-length` adds the block hash and iteration count.